nautobot/core/templates/inc/nav_menu.html

```diff
@@ -18,17 +18,17 @@
             {% if request.user.is_authenticated %}
                 <ul class="nav navbar-nav">
                     {% for tab_name, tab_details in registry.nav_menu.tabs.items %}
-                        {% if request.user|has_one_or_more_perms:tab_details.permissions or not "HIDE_RESTRICTED_UI"|settings_or_config %}
+                        {% if request.user|has_one_or_more_perms:tab_details.permissions or request.user|should_show_restricted_ui %}
                             <li class="dropdown">
                                 <a href="#" class="dropdown-toggle" data-tab-weight="{{ tab_details.weight }}" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
                                     {{ tab_name }} <span class="caret"></span>
                                 </a>
                                 <ul class="dropdown-menu">
                                     {% for group_name, group_details in tab_details.groups.items %}
-                                        {% if request.user|has_one_or_more_perms:group_details.permissions or not "HIDE_RESTRICTED_UI"|settings_or_config %}
+                                        {% if request.user|has_one_or_more_perms:group_details.permissions or request.user|should_show_restricted_ui:group_details.admin_only %}
                                             <li class="dropdown-header" data-group-weight="{{ group_details.weight }}">{{ group_name }}</li>
                                             {% for item_link, item_details in group_details.items.items %}
-                                                {% if request.user|has_one_or_more_perms:item_details.permissions or not "HIDE_RESTRICTED_UI"|settings_or_config %}
+                                                {% if request.user|has_one_or_more_perms:item_details.permissions or request.user|should_show_restricted_ui:group_details.admin_only %}
```

Comment:
Can we add some tests with `HIDE_RESTRICTED_UI` to confirm that the menu item is correctly shown/hidden?